web/massastation/src/store/networkStore.ts

```diff
@@ -1,16 +1,26 @@
 // EXTERNALS
 import { persist } from 'zustand/middleware';
+import type { NetworkInfo } from '@/models/NetworkModel';
 
 export interface NetworkStoreState {
   currentNetwork: string | null;
   setCurrentNetwork: (nickname: string | null) => void;
+  availableNetworks: NetworkInfo[];
+  setAvailableNetworks: (networks: NetworkInfo[]) => void;
+  getChainId: () => bigint | undefined;
 }
 
 const networkStore = persist<NetworkStoreState>(
-  (set) => ({
+  (set, get) => ({
     currentNetwork: null,
     setCurrentNetwork: (nickname: string | null) =>
       set({ currentNetwork: nickname }),
+    availableNetworks: [],
+    setAvailableNetworks: (networks: NetworkInfo[]) => set({ availableNetworks: networks }),
+    getChainId: (): bigint | undefined => {
+      const tempChId = get().availableNetworks.find((n: NetworkInfo) => n.name === get().currentNetwork)?.chainId;
+      return tempChId ? BigInt(tempChId) : undefined;
+    },
```

Comment:
imho it would have been cleaner to have a "currentChainId" property that is set in the `setCurrentNetwork`